dockerfiles/Dockerfile.training

```diff
@@ -0,0 +1,195 @@
+    mv ${MOFED_TARNAME} mofed-${MOFED_VERSION} &&\
+    cd mofed-${MOFED_VERSION} && \
+    ./mlnxofedinstall --basic --force --user-space-only --without-fw-update
+
+# build ucx suite (leverages mellanox libraries)
+# note: openmpi will not select ucx without multithreading enabled
+ARG UCX_VERSION
+ARG UCX_TARNAME=ucx-$UCX_VERSION
+ARG UCX_URL=https://github.com/openucx/ucx/releases/download/v${UCX_VERSION}/${UCX_TARNAME}.tar.gz
+RUN apt-get -y update && apt-get -y --no-install-recommends install \
+        libnuma-dev &&\
+    cd /stage && curl -fSsL ${UCX_URL} | tar xzf - &&\
+    cd ${UCX_TARNAME} &&\
+    ./configure \
+	--prefix=/opt/ucx \
+        --with-cuda=/usr/local/cuda \
+        --with-verbs=/usr \
+        --enable-mt &&\
+    make -j"$(nproc)" &&\
+    make install
+
+# build openmpi (use --prefix /opt/openmpi-xxx to move to runtime image)
+# note: require --enable-orterun-prefix-by-default for Azure machine learning compute
+# note: disable verbs as we use ucx middleware and don't want btl openib warnings
+ARG OPENMPI_VERSION
+ARG OPENMPI_TARNAME=openmpi-${OPENMPI_VERSION}
+ARG OPENMPI_URL=https://download.open-mpi.org/release/open-mpi/v%OMPI_BASE%/${OPENMPI_TARNAME}.tar.gz
+RUN export OMPI_BASE=${OPENMPI_VERSION%.*} &&\
+    cd /stage && curl -fSsL `echo ${OPENMPI_URL} | sed s/%OMPI_BASE%/$OMPI_BASE/` | tar xzf - &&\
+    cd ${OPENMPI_TARNAME} &&\
+    ./configure \
+        --prefix=/opt/openmpi-${OPENMPI_VERSION} \
+        --with-ucx=/opt/ucx \
+        --without-verbs \
+        --with-cuda=/usr/local/cuda \
+        --enable-mpirun-prefix-by-default \
+        --enable-orterun-prefix-by-default \
+        --enable-mca-no-build=btl-uct &&\
+    make -j"$(nproc)" install &&\
+    ldconfig
+ENV PATH=/opt/openmpi-${OPENMPI_VERSION}/bin:$PATH
+ENV LD_LIBRARY_PATH=/opt/openmpi-${OPENMPI_VERSION}/lib:$LD_LIBRARY_PATH
+
+# install mpi4py (be sure to link existing /opt/openmpi-xxx)
+RUN CC=mpicc MPICC=mpicc pip install mpi4py --no-binary mpi4py
+
+# build pytorch (using v1.6.0-rc3)
+# todo: need to update to pytorch 1.6.0 releaase when ready
+ENV CUDA_HOME="/usr/local/cuda" \
+    CUDNN_LIBRARY="/usr/lib/x86_64-linux-gnu" \
+    NCCL_INCLUDE_DIR="/usr/include" \
+    NCCL_LIB_DIR="/usr/lib/x86_64-linux-gnu" \
+    USE_SYSTEM_NCCL=1 
+RUN apt-get -y update && apt-get -y --no-install-recommends install git &&\
+    cd /stage && git clone https://github.com/pytorch/pytorch.git &&\
+    cd pytorch &&\
+    git checkout v1.6.0-rc3 &&\
+    git submodule update --init --recursive &&\
+    python setup.py bdist_wheel -d build/wheel &&\
+    pip install build/wheel/*.whl
+
+# build onnxruntime wheel with cuda and mpi support
+RUN cd /stage && git clone https://github.com/microsoft/onnxruntime.git &&\
+    cd onnxruntime &&\
+    git checkout v1.4.0 &&\
+    python tools/ci_build/build.py \
+        --cmake_extra_defines \
+            ONNXRUNTIME_VERSION=`cat ./VERSION_NUMBER` \
+        --config RelWithDebInfo \
```

Comment:
Is it necessary to use RelWithDebInfo? Or should Release be used here?

Reply:
Mm, I guess we should use Release.